.github/workflows/pyupgrade.yml

```
@@ -13,24 +13,41 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v4
 
+      # Step 2: Set up Python
       - name: Set up Python
         uses: actions/setup-python@v4
         with:
           python-version: "3.12"
 
+      # Step 3: Cache Poetry Dependencies
+      - name: Cache Poetry dependencies
+        uses: actions/cache@v2
```

actionlint (1.7.4)

the runner of "actions/cache@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)